ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -251,6 +259,10 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			if err != nil && !errors.IsNotFound(err) {
 				return ctrl.Result{RequeueAfter: RayJobDefaultRequeueDuration}, nil
 			}
+			_, err = r.deleteK8sJob(ctx, rayJobInstance)
```

Comment:
Why do we need to delete the Kubernetes Job here if we have set the controller reference (RayCluster is the owner and Job is the dependent)? See https://kubernetes.io/docs/concepts/architecture/garbage-collection/ for more details.

Reply:
Yeah, I got the ownership wrong.  RayCluster should be the owner like you said; I'll fix this and remove the deletion

Reply:
Fixed 471f979